File: js-demo/ollama_rest_api.js
const fs = require('fs');

/**
 * 使用Node.js通过HTTP请求与Ollama的qwen3:8b模型进行流式交互
 */
async function chatWithQwen3Stream(prompt, model = 'qwen3:8b') {
  const url = 'http://localhost:11434/api/generate';
  
  const payload = {
    model: model,
    prompt: prompt,
    stream: true  // 设置为true以启用流式输出
  };
  
  try {
    const response = await fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(payload)
    });
    
    if (response.ok) {
      // 获取响应流
      const reader = response.body;
      
      // 读取流数据
      for await (const chunk of reader) {
        const decodedChunk = new TextDecoder().decode(chunk);
        // Ollama流式响应每行是一个独立的JSON对象
        const lines = decodedChunk.split('\n').filter(line => line.trim() !== '');
        
        for (const line of lines) {
          try {
            const jsonData = JSON.parse(line);
            if (jsonData.response) {
              process.stdout.write(jsonData.response);
            }
          } catch (parseError) {
            // 忽略解析错误
          }
        }
      }
    } else {
      console.error(`Error: ${response.status} - ${response.statusText}`);
      return null;
    }
  } catch (error) {
    console.error('Error connecting to Ollama:', error);
    return null;
  }
}

/**
 * 使用Node.js通过HTTP请求与Ollama的qwen3:8b模型进行交互（非流式）
 */
async function chatWithQwen3(prompt, model = 'qwen3:8b') {
  const url = 'http://localhost:11434/api/generate';
  
  const payload = {
    model: model,
    prompt: prompt,
    stream: false  // 设置为false以获得完整响应
  };
  
  try {
    const response = await fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(payload)
    });
    
    if (response.ok) {
      const data = await response.json();
      return data.response;
    } else {
      console.error(`Error: ${response.status} - ${response.statusText}`);
      return null;
    }
  } catch (error) {
    console.error('Error connecting to Ollama:', error);
    return null;
  }
}

// 主函数 - 演示如何使用
async function main() {
  // 示例1：向模型提问（流式输出）
  const question1 = "请用5句话描述人工智能的发展历程。";
  console.log(`问题: ${question1}`);
  process.stdout.write("回答: ");
  
  await chatWithQwen3Stream(question1);
  
  console.log('\n\n---另一个示例（非流式）---');
  
//   // 示例2：编程相关问题
//   const question2 = "请用5句话描述人工智能的发展历程。";
//   console.log(`问题: ${question2}`);
  
//   const answer2 = await chatWithQwen3(question2);
//   if (answer2) {
//     console.log(`回答: ${answer2}`);
//   }
}

// 运行主函数
main();

File: python-demo/ollama_rest_api.py
import requests
import json

def chat_with_qwen3_stream(prompt, model="qwen3:8b"):
    """
    使用REST API与Ollama的qwen3:8b模型进行流式交互
    """
    # Ollama API端点
    url = "http://localhost:11434/api/generate"
    
    # 请求数据
    payload = {
        "model": model,
        "prompt": prompt,
        "stream": True  # 设置为True以启用流式输出
    }
    
    # 发送POST请求并流式处理响应
    with requests.post(url, json=payload, stream=True) as response:
        # 检查响应状态
        if response.status_code == 200:
            # 流式读取响应内容
            for line in response.iter_lines():
                if line:
                    decoded_line = line.decode('utf-8')
                    try:
                        json_data = json.loads(decoded_line)
                        # 输出模型生成的文本片段
                        if 'response' in json_data:
                            print(json_data['response'], end='', flush=True)
                    except json.JSONDecodeError:
                        pass
        else:
            print(f"Error: {response.status_code} - {response.text}")

def chat_with_qwen3(prompt, model="qwen3:8b"):
    """
    使用REST API与Ollama的qwen3:8b模型进行交互（非流式）
    """
    # Ollama API端点
    url = "http://localhost:11434/api/generate"
    
    # 请求数据
    payload = {
        "model": model,
        "prompt": prompt,
        "stream": False  # 设置为False以获得完整响应
    }
    
    # 发送POST请求
    response = requests.post(url, json=payload)
    
    # 检查响应状态
    if response.status_code == 200:
        data = response.json()
        return data.get("response", "")
    else:
        print(f"Error: {response.status_code} - {response.text}")
        return None

if __name__ == "__main__":
    # 示例：向模型提问（流式输出）
    question = "请写一首关于春天的诗，要求至少4行。"
    print(f"问题: {question}")
    print("回答: ", end='')
    
    chat_with_qwen3_stream(question)
    
    # print("\n\n---另一个示例（非流式）---")
    # question = "请写一首关于春天的诗，要求至少4行。"
    # print(f"问题: {question}")
    
    # answer = chat_with_qwen3(question)
    # if answer:
    #     print(f"回答: {answer}")